### PR TITLE
New cop Lint/RescueType

### PR DESCRIPTION
**Pull request by rrosenblum**
This came up some time ago (I think when we created `Lint/ShadowedException`) and seemed useful to have. I was also considering the name `RescueArgumentType`.

Back in Ruby 1.8.7, `rescue nil` would effectively disable the `rescue`.

### CODE REVIEW

File: spec/rubocop/cop/lint/rescue_type_spec.rb [hunk 1]
```diff
@@ -0,0 +1,267 @@
+# frozen_string_literal: true
+
+require 'pry'
```
Comment (Drenmi) [inline spec/rubocop/cop/lint/rescue_type_spec.rb]: This might not be needed anymore. 🙂

Comment (rrosenblum) [inline spec/rubocop/cop/lint/rescue_type_spec.rb]: I removed it and tried adding a `binding.pry`, it did not work. 🙁

File: lib/rubocop/cop/lint/rescue_type.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      #   # good
+      #   def foo
+      #     bar
+      #   rescue NameError
+      #     baz
+      #   end
+      class RescueType < Cop
+        include RescueNode
+
+        MSG = 'Do not rescue from `%s`. ' \
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/rescue_type.rb]: I'd convert this to a single sentence for consistency with other messages we've got.

**Comment by rrosenblum** [comment]
The message has been changed to a single sentence and the code has been rebased.

**Comment by bbatsov** [comment]
The build is failing.

**Pull request merged**